ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -326,12 +382,19 @@ func ConfigureSeGroup(client *clients.AviClient, seGroup *models.ServiceEngineGr
 	return true
 }
 
-func (a *AviControllerInfra) AnnotateSystemNamespace(seGroup string, cloudName string) bool {
+func (a *AviControllerInfra) AnnotateSystemNamespace(seGroup string, cloudName string, retries ...int) bool {
+	retryCount := 0
+	if len(retries) > 0 {
+		retryCount = retries[0]
+	}
+	if retryCount > 3 {
+		utils.AviLog.Fatalf("Failed to Annotate the %s namespace, shutting down", utils.GetAKONamespace())
```

Comment:
with older behaviour, if annotation fails AKO will be in running state and yet won't process any ingress/gateway. Restarting ensures that Annotation keeps getting retried and if there is a consistent failure, the AKO deployment goes to CLBO - reflecting the failure state instead of running state